src/Forcings/advective_forcing.jl

```diff
@@ -1,10 +1,18 @@
 using Oceananigans.Advection: div_Uc, div_𝐯u, div_𝐯v, div_𝐯w
 using Oceananigans.Fields: ZeroField, ConstantField
 using Oceananigans.Utils: sum_of_velocities
+using Oceananigans.BoundaryConditions: OpenBoundaryCondition, FieldBoundaryConditions
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryCondition
 using Adapt
 
-maybe_constant_field(u) = u
-maybe_constant_field(u::Number) = ConstantField(u)
+maybe_field(u, args...) = u
+
+function maybe_field(u::Number, loc, grid, u_bcs)
```

Comment:
is this different from the function `field`?

Reply:
It is if `u` isn't a Number. (For example and array or a `ConstantField`.)
There's a slight difference for numbers as well since `u_bcs` isn't ` FieldBoundaryConditions`. (I'm trying to keep the interface simple and avoid having the user pass a whole set of `FieldBoundaryConditions` when creating a simple slip velocity forcing.

Reply:
> Don't you need to pass boundary conditions to the Field when creating it? Because if so, that's what this code does (I simplified it a bit more if you wanna take another look).

Yes, `Field` need boundary conditions if they are not default. For velocities, I would:

1. Use default velocity boundary conditions if users give a Number (eg impenetrable)
2. Also support explicitly providing `Field` which bundle boundary conditions